pkg/project/filter.go

```diff
@@ -28,6 +29,10 @@ func WithFiltering(config FilterConfig) heartbeat.HandleOption {
 				if err != nil {
 					log.Debugln(err.Error())
 
+					if h.LocalFileTemporary {
```

Comment:
It already calls `remote.WithCleanup()` at the end of the chain, so is it needed?

Reply:
That's make more sense to me.

Reply:
I added `IsFiltered` and it's working well.

Reply:
I had to revert back to removing filtered heartbeats from the array, because it causes unintended side-effects like downloading filtered remote files, calculating file stats on filtered files, detecting language of filtered files, etc.

Removing the heartbeat is better than checking `heartbeat.IsFiltered` in every heartbeat handler.